lib/fog/vcloud_director/requests/compute/get_edge_gateway.rb

```diff
@@ -17,34 +18,34 @@ def get_edge_gateway(id)
               :expects => 200,
               :idempotent => true,
               :method => 'GET',
-              :parser => Fog::ToHashDocument.new,
+              :parser => Fog::Parsers::Compute::VcloudDirector::EdgeGatewayConfiguration.new,
               :path => "admin/edgeGateway/#{id}"
           )
 
-          ensure_list! response.body[:Configuration], :GatewayInterfaces, :GatewayInterface
```

Comment:
maybe you just delete the code, it is already on github. Apply this to the rest of the code